core/src/main/java/org/apache/brooklyn/core/mgmt/persist/FileBasedObjectStore.java

```diff
@@ -346,51 +345,17 @@ protected File backupDirByMoving(File dir) throws InterruptedException, IOExcept
         return newDir;
     }
 
-    private static boolean WARNED_ON_NON_ATOMIC_FILE_UPDATES = false; 
     /** 
      * Attempts an fs level atomic move then fall back to pure java rename.
      * Assumes files are on same mount point.
-     * <p>
-     * TODO Java 7 gives an atomic Files.move() which would be preferred.
+     * Overwriting existing destFile
      */
     static void moveFile(File srcFile, File destFile) throws IOException, InterruptedException {
-        // Try rename first - it is a *much* cheaper call than invoking a system call in Java. 
-        // However, rename is not guaranteed cross platform to succeed if the destination exists,
-        // and not guaranteed to be atomic, but it usually seems to do the right thing...
-        boolean result;
-        result = srcFile.renameTo(destFile);
-        if (result) {
-            if (log.isTraceEnabled()) log.trace("java rename of {} to {} completed", srcFile, destFile);
-            return;
-        }
-        
-        if (!Os.isMicrosoftWindows()) {
-            // this command, if it succeeds, is guaranteed to be atomic, and it will usually overwrite
-            String cmd = "mv '"+srcFile.getAbsolutePath()+"' '"+destFile.getAbsolutePath()+"'";
-            
-            int exitStatus = new ProcessTool().execCommands(MutableMap.<String,String>of(), MutableList.of(cmd), null);
-            // prefer the above to the below because it wraps it in the appropriate bash
-//            Process proc = Runtime.getRuntime().exec(cmd);
-//            result = proc.waitFor();
-            
-            if (log.isTraceEnabled()) log.trace("FS move of {} to {} completed, code {}", new Object[] { srcFile, destFile, exitStatus });
-            if (exitStatus == 0) return;
+        if (destFile.exists()) {
```

Comment:
I understand. about the exception being thrown... if we have any performance issues because of that we can change it then.. but I don't think so..